core/src/io/anuke/mindustry/world/BaseBlock.java

```diff
@@ -33,6 +33,8 @@
     public float liquidFlowFactor = 4.9f;
     public float powerCapacity = 10f;
 
+    public float powerProduction;
```

Comment:
Why is this in `BaseBlock` when it's generator-specific?

Reply:
Where should I put it then?

Reply:
Put it in `GeneratorBlock`, with a `float getPowerProduction(Tile tile)` in `BaseBlock` that calculates the current block power production, returning 0 by default and `powerProduction * entity.productionEfficiency` for generators.

Reply:
Makes sense, but the problem with that is now I'll have to change `DebugBlocks` because `powerInfinite` inherits from `PowerNode` which is not a `PowerGenerator`, thus doesn't have the `powerProduction` attribute. Is it ok if I change it to a `PowerGenerator`?

Reply:
There's also a problem with `FusionReactor`. Its entity is a `GenericCrafterEntity`. I'd have to turn `GenericCrafterEntity` and `GeneratorEntity` into interfaces or something.

Reply:
This is why I suggested making power production a method in `BaseBlock`. `FusionReactor` and `powerInfinite` can override it.